src/lib/extension-load-queue.ts

```diff
@@ -81,27 +169,44 @@ export const createLoadQueue = <T>({
             return next();
         }
 
+        // Mark the processing as complete
         processing = false;
     }
 
+    // Start the queue processing
     unqueue();
 };
 
-const noop = () => {};
-
-export function useController<T>(store: ScriptStore<T>) {
+/**
+ * A no-operation function used as a default cleanup function.
+ *
+ * @returns {void}
+ */
+const noop = (): void => {};
+
+/**
+ * React hook to manage and use a controller with a script store.
+ *
+ * @param {ScriptStore<T>} store - The store where the controller is managed.
+ * @returns {T | null} The current controller or null if not available.
+ */
+export function useController<T>(store: ScriptStore<T>): T | null {
     const [controller, setController] = useState<T | null>(null);
 
     useEffect(() => {
         if (store) {
-            store.push(setController);
+            store.push(setController); // Add setController to the store
 
             return () => {
                 const index = store.indexOf(setController);
                 if (index > -1) {
+                    // Remove setController when unmounting
                     store.splice(index, 1);
                 }
             };
+        } else {
+            // eslint-disable-next-line no-console
+            console.warn('Store is not provided to useController'); // Replace console.warn with a logging function if necessary
```

Comment:
I'd say just throwing makes more sense here, maybe even removing the `if (store)` check altogether, since `store` cannot be nullable judging from the argument types